api/grpc_server.go

```diff
@@ -321,25 +322,41 @@ func (s SpacemeshGrpcService) SetLoggerLevel(ctx context.Context, msg *pb.SetLog
 
 func (s SpacemeshGrpcService) GetAccountTxs(ctx context.Context, txsSinceLayer *pb.GetTxsSinceLayer) (*pb.AccountTxs, error) {
 	log.Info("GRPC GetAccountTxs msg")
-	acc := types.HexToAddress(txsSinceLayer.Account.Address)
-	if txsSinceLayer.StartLayer > uint64(s.Tx.LatestLayer()) {
+
+	currentPBase := s.Tx.ValidatedLayer()
+
+	addr := types.HexToAddress(txsSinceLayer.Account.Address)
+	minLayer := types.LayerID(txsSinceLayer.StartLayer)
+	if minLayer > s.Tx.LatestLayer() {
 		return &pb.AccountTxs{}, fmt.Errorf("invalid start layer")
 	}
-	var allTxs []types.TransactionId
-	for i := txsSinceLayer.StartLayer; i < uint64(s.Tx.LatestLayer()); i++ {
-		txs := s.Tx.GetTransactionsByDestination(types.LayerID(i), acc)
-		allTxs = append(allTxs, txs...)
-		moreTxs := s.Tx.GetTransactionsByOrigin(types.LayerID(i), acc)
-		allTxs = append(allTxs, moreTxs...)
+
+	txs := pb.AccountTxs{ValidatedLayer: currentPBase.Uint64()}
```

Comment:
not that important, but you can pre allocate the txs.Txs to the relevant size and avoid extra memory allocations

Reply:
I considered it, but it would make the code slightly more convoluted (I'd need to first generate the two lists, then allocate, then copy items from both lists) and the performance impact should be minimal (allocation doesn't happen for each appended item - Go doubles the underlying array every time it runs out of space, so `log(n)` times).

I think it's not worth it for this function, can reconsider if/when we decide it needs a performance optimization.